src/main/kotlin/net/atos/zac/signalering/SignaleringService.kt

```diff
@@ -176,15 +190,35 @@ class SignaleringService @Inject constructor(
             .resultList
     }
 
-    fun latestSignalering(parameters: SignaleringZoekParameters): ZonedDateTime? {
+    private fun listSignaleringen(
+        parameters: SignaleringZoekParameters,
+        pageNumber: Int,
+        pageSize: Int
+    ): List<Signalering> {
+        val builder = entityManager.criteriaBuilder
+        val query = builder.createQuery(
+            Signalering::class.java
+        )
+        val root = query.from(Signalering::class.java)
+        return entityManager.createQuery(
+            query.select(root)
+                .where(getSignaleringWhere(parameters, builder, root))
+                .orderBy(builder.desc(root.get<Any>("tijdstip")))
+        )
+            .setFirstResult(pageNumber * pageSize)
+            .setMaxResults(pageSize)
+            .resultList
+    }
+
+    fun latestSignalering(): ZonedDateTime? {
```

Comment:
Why can it return null?

So this function returns a ZonedDateTime. That is not what I would expect from a function called `latestSignalering`... If I see a function named `latestSignalering` I expect it to return the latest signalering, not some date time.  Maybe rename the function?

Reply:
I agree. Renamed to `latestSignaleringOccurrence`